services/crates/crates-dependents.service.js

```diff
@@ -0,0 +1,49 @@
+import Joi from 'joi'
+import { pathParams } from '../index.js'
+import { metric } from '../text-formatters.js'
+import { nonNegativeInteger } from '../validators.js'
+import { BaseCratesService, description } from './crates-base.js'
+
+const dependentsResponseSchema = Joi.object({
+  meta: Joi.object({
+    total: nonNegativeInteger,
+  }).required(),
+}).required()
+
+export default class CratesDependents extends BaseCratesService {
+  static category = 'other'
+  static route = { base: 'crates/dependents', pattern: ':crate' }
```

Comment:
If you have a look over the other rust crates badges, we accept an optional `:version?` param which allows us to present the statistic for a specific version.

Could we also do that here, or does this `reverse_dependencies` endpoint not allow us to do that in this case?

Reply:
The API does not have any documentation to begin with. I tried `GET`ting these URLs with `tokio` and `1.39.2` version but only got 404.

 - `https://crates.io/api/v1/crates/tokio/v1.39.2/reverse_dependencies/`
 - `https://crates.io/api/v1/crates/tokio/1.39.2/reverse_dependencies/`
 - `https://crates.io/api/v1/crates/tokio/reverse_dependencies/v1.39.2/`
 - `https://crates.io/api/v1/crates/tokio/reverse_dependencies/1.39.2/`

While there's no documentation for the API, there's actually a crate for the API. [This struct](https://docs.rs/crates_io_api/latest/crates_io_api/struct.ReverseDependencies.html) hints no version at all, so it's almost guaranteed to say crates.io does not support versioning with reverse dependencies.

Reply:
OK, cool. No worries. Thanks for looking into it.